docs/db_migration_guide.md

```diff
@@ -0,0 +1,244 @@
+    style M1_Postgres fill:#cce5ff,stroke:#004085
+    style M2_Postgres fill:#cce5ff,stroke:#004085
+    style M3_Postgres fill:#cce5ff,stroke:#004085
+
+```
+
+- **Stage 1**: Migrate from bbolt to a SQL-based **kvdb** backend using the [lndinit](https://github.com/lightninglabs/lndinit/blob/main/docs/data-migration.md) tool.
+- **Stage 2**: Incrementally migrate subsystem data (invoices, graph, payments, etc.) from **kvdb** to SQL native **relational tables** as support becomes available.
+
+---
+
+## Stage 1: Migration from bbolt to SQLite/Postgres (kvdb)
+
+LND cannot run with mixed backends, so all users must first leave bbolt behind. This stage uses the `lndinit` utility to perform an offline migration.
+
+### Choosing Your Target Backend
+
+| Backend   | Performance (in kvdb mode) | Default in LND? | Long-Term Viability |
+|-----------|--------------------------|------------------|----------------------|
+| Postgres  | Mediocre                 | No               | ✅ |
+| SQLite    | Good                     | **Yes (future)** | ✅ |
+
+> 💡 **Recommendation**: Unless you require Postgres for infrastructure reasons, **migrate to SQLite kvdb** as your interim backend.
+
+### Postgres kvdb Migration
+
+A migration script is available via `lndinit`:
+
+- [Postgres migration script](https://github.com/lightninglabs/lndinit/blob/main/docs/data-migration.md#using-postgres-as-the-destination-remote-database)
+- **Caveat**: Users report degraded performance in kvdb mode. Only proceed if you plan to **immediately follow with Stage 2** to migrate the available data stores to relational DB.
+This will mitigate the poor Postgres performance on kvdb.
+
+### SQLite kvdb Migration
+
+- [SQLite migration script](https://github.com/lightninglabs/lndinit/blob/main/docs/data-migration.md#using-sqlite-as-the-destination-remote-database)
+- **Advantage**: Maintains good performance while waiting for full relational migration.
+
+---
+
+## Stage 2: Migration from kvdb to Relational Database
+
+This stage unlocks true SQL performance by restructuring data into relational tables. Migration is **per-subsystem** and **incremental**.
+
+### Subsystem Readiness
+
+| Subsystem           | Relational Backend | Migration Script | Status |
+|---------------------|--------------------|------------------|--------|
+| Invoices            | ✅ Available        | ✅              | Available with **v0.19** |
+| Graph               | ✅ Available        | ✅              | Available with **v0.20** |
+| Payments            | 🚧 In Progress      | Planned          | Targeted with **v0.21**|
+| Btcwallet           | 🚧 In Progress      | Planned          | Targeted with **v0.21**|
+| Forwarding History  | ❌ TBD              | ❌ TBD            | Future work |
+
+---
+
+## Known Limitations and Edge Cases
+
+- **Single database engine required**: LND requires a single consistent backend.
+You cannot run invoices in relational mode while payments remain in kvdb *unless*
```

Comment:
maybe use a real example like invoices & graph (since the reference to payments would at this stage be confusing)